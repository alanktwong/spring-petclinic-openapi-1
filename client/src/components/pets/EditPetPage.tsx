import * as React from 'react';

import { RouteComponentProps, withRouter } from 'react-router-dom';

import { ISelectOption } from '../../types';

import { Owner, Pet, PetApi } from 'petclinic-api';

import LoadingPanel from './LoadingPanel';
import PetEditor from './PetEditor';

import createPetEditorModel from './createPetEditorModel';

interface IEditPetPageProps extends RouteComponentProps {}

interface IEditPetPageState {
  pet?: Pet;
  owner?: Owner;
  petTypes?: ISelectOption[];
}

class EditPetPage extends React.Component<
  IEditPetPageProps,
  IEditPetPageState
> {
  componentDidMount() {
    const ownerId = Number(this.props.match.params['ownerId']);
    const petId = Number(this.props.match.params['petId']);

    const loadPetPromise = new PetApi().getPet({ ownerId, petId });

    createPetEditorModel(ownerId, loadPetPromise).then(model =>
      this.setState(model)
    );
  }

  render() {
    if (!this.state) {
      return <LoadingPanel />;
    }

    const { pet, owner, petTypes } = this.state;

    if (!pet) {
      return <LoadingPanel />;
    }
    if (!owner) {
      return <LoadingPanel />;
    }
    if (!petTypes) {
      return <LoadingPanel />;
    }

    return <PetEditor pet={pet} owner={owner} petTypes={petTypes} />;
  }
}

export default withRouter(EditPetPage);
