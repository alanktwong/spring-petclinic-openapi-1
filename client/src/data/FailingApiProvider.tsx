import * as React from 'react';
import { FailingApi } from 'petclinic-api';
import { Subtract } from 'utility-types';

export interface WithFailingApiProps {
  failingApi: FailingApi;
}

export const FailingApiContext = React.createContext(new FailingApi());

export const withFailingApi = <P extends WithFailingApiProps>(
  Component: React.ComponentType<P>
) =>
  class WithFailingApi extends React.Component<
    Subtract<P, WithFailingApiProps>
  > {
    static contextType = FailingApiContext;
    context!: React.ContextType<typeof FailingApiContext>;

    render() {
      return <Component failingApi={this.context} {...(this.props as P)} />;
    }
  };
