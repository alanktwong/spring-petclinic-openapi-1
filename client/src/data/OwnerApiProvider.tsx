import * as React from 'react';
import { OwnerApi } from 'petclinic-api';
import { Subtract } from 'utility-types';

export interface WithOwnerApiProps {
  ownerApi: OwnerApi;
}

export const OwnerApiContext = React.createContext(new OwnerApi());

export const withOwnerApi = <P extends WithOwnerApiProps>(
  Component: React.ComponentType<P>
) =>
  class WithOwnerApi extends React.Component<Subtract<P, WithOwnerApiProps>> {
    static contextType = OwnerApiContext;
    context!: React.ContextType<typeof OwnerApiContext>;

    render() {
      return <Component ownerApi={this.context} {...(this.props as P)} />;
    }
  };
