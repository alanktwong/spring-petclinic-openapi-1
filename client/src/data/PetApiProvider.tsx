import * as React from 'react';
import { PetApi } from 'petclinic-api';
import { Subtract } from 'utility-types';

export interface WithPetApiProps {
  petApi: PetApi;
}

export const PetApiContext = React.createContext(new PetApi());

export const withPetApi = <P extends WithPetApiProps>(
  Component: React.ComponentType<P>
) =>
  class WithPetApi extends React.Component<Subtract<P, WithPetApiProps>> {
    static contextType = PetApiContext;
    context!: React.ContextType<typeof PetApiContext>;

    render() {
      return <Component petApi={this.context} {...(this.props as P)} />;
    }
  };
