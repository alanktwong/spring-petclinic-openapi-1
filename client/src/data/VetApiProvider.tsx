import * as React from 'react';
import { VetApi } from 'petclinic-api';
import { Subtract } from 'utility-types';

export interface WithVetApiProps {
  vetApi: VetApi;
}

export const VetApiContext = React.createContext(new VetApi());

export const withVetApi = <P extends WithVetApiProps>(
  Component: React.ComponentType<P>
) =>
  class WithVetApi extends React.Component<Subtract<P, WithVetApiProps>> {
    static contextType = VetApiContext;
    context!: React.ContextType<typeof VetApiContext>;

    render() {
      return <Component vetApi={this.context} {...(this.props as P)} />;
    }
  };
