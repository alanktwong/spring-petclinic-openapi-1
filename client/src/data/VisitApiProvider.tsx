import * as React from 'react';
import { VisitApi } from 'petclinic-api';
import { Subtract } from 'utility-types';

export interface WithVisitApiProps {
  visitApi: VisitApi;
}

export const VisitApiContext = React.createContext(new VisitApi());

export const withVisitApi = <P extends WithVisitApiProps>(
  Component: React.ComponentType<P>
) =>
  class WithVisitApi extends React.Component<Subtract<P, WithVisitApiProps>> {
    static contextType = VisitApiContext;
    context!: React.ContextType<typeof VisitApiContext>;

    render() {
      return <Component visitApi={this.context} {...(this.props as P)} />;
    }
  };
