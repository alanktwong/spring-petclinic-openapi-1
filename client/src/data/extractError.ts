import { IError } from '../types';

const extractError = (response): Promise<IError> => {
  if (response.json) {
    return response.json();
  } else {
    return Promise.resolve({ message: `${response}` });
  }
};

export default extractError;
