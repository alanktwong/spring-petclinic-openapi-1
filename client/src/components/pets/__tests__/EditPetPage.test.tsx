import React from 'react';
import renderer from 'react-test-renderer';
import EditPetPage from '../EditPetPage';
import waitForExpect from 'wait-for-expect';
import { OwnerApiContext } from '../../../data/OwnerApiProvider';
import { PetApiContext } from '../../../data/PetApiProvider';
import { OwnerApi, PetApi } from 'petclinic-api';
import { Route, StaticRouter } from 'react-router-dom';

it('renders correctly', async () => {
  const ownerMock = new OwnerApi();
  ownerMock.getOwner = async () => ({
    id: 1,
    firstName: 'firstName1',
    lastName: 'lastName1',
    address: 'address1',
    city: 'city1',
    telephone: '012345',
    pets: [],
  });

  const petType = {
    id: 3,
    name: 'typeName3',
  };

  const petMock = new PetApi();
  petMock.listPetTypes = async () => [petType];
  petMock.getPet = async () => ({
    id: 2,
    name: 'petName2',
    birthDate: new Date(2019, 0, 1),
    typeId: 3,
    type: petType,
    visits: [],
  });

  const component = renderer.create(
    <StaticRouter location="/owners/1/pets/2/edit">
      <PetApiContext.Provider value={petMock}>
        <OwnerApiContext.Provider value={ownerMock}>
          <Route
            path="/owners/:ownerId(\d+)/pets/:petId(\d+)/edit"
            component={EditPetPage}
          />
        </OwnerApiContext.Provider>
      </PetApiContext.Provider>
    </StaticRouter>
  );

  const page = component.root.find(node => node.type.name === 'EditPetPage');

  await waitForExpect(() => {
    expect(page.instance.state.data.owner.id).toBe(1);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
