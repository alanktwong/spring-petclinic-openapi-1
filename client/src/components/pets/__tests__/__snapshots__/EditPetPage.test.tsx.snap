// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h2>
    Update Pet
  </h2>
  <form
    className="form-horizontal"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="form-group has-feedback"
    >
      <div
        className="form-group"
      >
        <label
          className="col-sm-2 control-label"
        >
          Owner
        </label>
        <div
          className="col-sm-10"
        >
          firstName1
           
          lastName1
        </div>
      </div>
      <div
        className="form-group "
      >
        <label
          className="col-sm-2 control-label"
          htmlFor="name"
        >
          Name
        </label>
        <div
          className="col-sm-10"
        >
          <input
            className="form-control"
            id="name"
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            value="petName2"
          />
          <span
            aria-hidden="true"
            className="glyphicon glyphicon-ok form-control-feedback"
          />
        </div>
      </div>
      <div
        className="form-group "
      >
        <label
          className="col-sm-2 control-label"
          htmlFor="birthDate"
        >
          Birth date
        </label>
        <div
          className="col-sm-10"
        >
          <div
            className="react-datepicker-wrapper"
          >
            <div
              className="react-datepicker__input-container"
            >
              <input
                className="form-control"
                disabled={false}
                id="birthDate"
                name="birthDate"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                readOnly={false}
                type="text"
                value="2019-01-01"
              />
            </div>
          </div>
          <span
            aria-hidden="true"
            className="glyphicon glyphicon-ok form-control-feedback"
          />
        </div>
      </div>
      <div
        className="form-group "
      >
        <label
          className="col-sm-2 control-label"
          htmlFor="typeId"
        >
          Type
        </label>
        <div
          className="col-sm-10"
        >
          <select
            className="form-control"
            name="typeId"
            onBlur={[Function]}
            onChange={[Function]}
            size={5}
            value={3}
          >
            <option
              value={3}
            >
              typeName3
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="form-group"
    >
      <div
        className="col-sm-offset-2 col-sm-10"
      >
        <button
          className="btn btn-default"
          disabled={false}
          type="submit"
        >
          Update Pet
        </button>
      </div>
    </div>
  </form>
</div>
`;
